akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1460,12 +1460,41 @@ trait FlowOps[+Out, +Mat] {
    *
    * '''Cancels when''' downstream cancels
    *
-   * @param seed Provides the first state for extrapolation using the first unconsumed element
    * @param extrapolate Takes the current extrapolation state to produce an output element and the next extrapolation
    *                    state.
    */
   def expand[U](extrapolate: Out ⇒ Iterator[U]): Repr[U] = via(new Expand(extrapolate))
 
+  /**
+   * Allows a faster downstream to progress of a slower upstream.
+   *
+   * This is achieved by introducing "extrapolated" elements - based on those from upstream - whenever downstream
+   * signals demand.
+   *
+   * Pressurize does not support [[akka.stream.Supervision.Restart]] and [[akka.stream.Supervision.Resume]].
+   * Exceptions from the `extrapolate` function will complete the stream with failure.
+   *
+   * '''Emits when''' downstream stops backpressuring, AND EITHER upstream emits OR initial element is present OR
+   * `extrapolate` is non-empty and applicable
+   *
+   * '''Backpressures when''' downstream backpressures or current `extrapolate` runs empty
+   *
+   * '''Completes when''' upstream completes and current `extrapolate` runs empty
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @param extrapolate takes the current upstream element and provides a sequence of "extrapolated" elements based
+   *                    on the original, to be emitted in case downstream signals demand
+   * @param initial the initial element to be emitted, in case upstream is able to stall the entire stream
+   */
+  def pressurize[U >: Out](extrapolate: U ⇒ Stream[U], initial: Option[U] = None): Repr[U] = {
```

Comment:
Ah, I see now why `expand` always passes the element through the function, the type signature is trickier/less useful this way with the lower type bound as it is easy to just widen it to anything. Very unsure how we can get this to work for Java without making the type parameters `Object` across the board as there is no such thing as lower type bounds in Java.

I think this should instead be called call `expandOrEmit` which would highlight the difference between this and the existing one, but, skip the initial as that can be achieved through composition (maybe mention in docs though), the function should not produce `Stream` as that is less generic, has memory leak issues, and limits use cases, instead it should produce an `Iterator` (possibly a custom this-value-and-then-the-rest-iterator of some kind for performance reasons).

Reply:
Regarding lower type bounds - not exactly, you can specify wildcard lower type bounds (`<? super Sth>`). It appears it's useless in our case anyway, i.e. the user effectively operates on `Object` anyway.

Re `Stream` - OK, so there's a reason to use that beyond the namespace collision. Could you elaborate on the memory leak issues, by the way? Do you mean the increased overhead as described [here](https://stackoverflow.com/questions/20360161/streamfilter-runs-out-of-memory-for-1-000-000-items), or something else?

Re `Iterator` - any specific concerns you're thinking about using a custom one? The Scala standard library's `Iterator.++` seems to be optimized for efficiency already. 

Re `expandOrEmit` - I'm not partial to names with logical operators for several reasons, the one most relevant here is that they inhibit recall of the semantics (since you have to remember two "subthings"). I'll switch to `extrapolate` on @ktoso 's suggestion (see below) for now.

Reply:
> Re Stream - OK, so there's a reason to use that beyond the namespace collision. Could you elaborate on the memory leak issues, by the way? Do you mean the increased overhead as described here, or something else?

it memorizes, and that's not something we need here ;) Naming things aside, it's not really a collection that is very useful unless you specifically know you want memoization